python/hsfs/feature_store.py

```diff
@@ -965,9 +965,15 @@ def get_or_create_spine_group(
     ):
         """Create a spine group metadata object.
 
-        Instead of using a feature group to save a label/prediction target, you can use a spine together with a dataframe containing the labels.
-        A Spine is essentially a metadata object similar to a feature group, however, the data is not materialized in the feature store.
-        It only containes the needed metadata such as the relevant event time column and primary key columns to perform point-in-time correct joins.
+        Instead of using a feature group to save a label/prediction target, you can use a spine together with a dataframe containing the labels and join keys for features in other feature groups.
+        A Spine is essentially a metadata object similar to a feature group, however, its data is not stored in the feature store.
+        The Spine stored in the feature store only contains the needed metadata such as the name, version, primary key column(s), and event time column. 
+        The Spine DataFrame is provided when you need to (1) create training data and (2) create batch inference data. The Spine DataFrame should also contain any join keys (primary keys to other feature groups) needed to join features included in a feature view containing the Spine group.If you don’t include the event_time in the Spine DataFrame (such as in batch inference), it will retrieve the latest feature value for that feature using the join key(s).
```

Comment:
Are you sure about the last part? `If you don’t include the event_time in the Spine DataFrame (such as in batch inference), it will retrieve the latest feature value for that feature using the join key(s).`

I don't think that's the case. if you don't provide the event_time, I think Hopsworks falls back to non-pit join and it doesn't guarantee time ordering. You most likely end up with multiple rows for each pk in the spine feature group (one for each event in the joined feature group).